include/linker/sections.h

```diff
@@ -14,21 +14,22 @@
 
 #ifndef ZEPHYR_INCLUDE_LINKER_SECTIONS_H_
 #define ZEPHYR_INCLUDE_LINKER_SECTIONS_H_
+#include <misc/util.h>
 
-#define _TEXT_SECTION_NAME text
-#define _RODATA_SECTION_NAME rodata
-#define _CTOR_SECTION_NAME ctors
-/* Linker issue with XIP where the name "data" cannot be used */
-#define _DATA_SECTION_NAME datas
-#define _BSS_SECTION_NAME bss
-#define _NOINIT_SECTION_NAME noinit
+#define _TEXT_SECTION_NAME .text
+#define _RODATA_SECTION_NAME .rodata
+#define _CTOR_SECTION_NAME .ctors
+#define _DATA_SECTION_NAME .data
+#define _BSS_SECTION_NAME .bss
+#define _NOINIT_SECTION_NAME .noinit
 
-#define _APP_SMEM_SECTION_NAME		app_smem
-#define _APP_DATA_SECTION_NAME		app_datas
-#define _APP_BSS_SECTION_NAME		app_bss
-#define _APP_NOINIT_SECTION_NAME	app_noinit
+#define _APP_SMEM_SECTION_NAME		.app_smem
+#define _APP_DATA_SECTION_NAME		.app_datas
+#define _APP_BSS_SECTION_NAME		.app_bss
+#define _APP_NOINIT_SECTION_NAME	.app_noinit
 
-#define _UNDEFINED_SECTION_NAME undefined
+#define _UNDEFINED_SECTION_NAME .undefined
+#define _TEXT_SECTION_NAME_2 UTIL_CAT(_TEXT_SECTION_NAME,.2)
```

Comment:
After having a second look. I suspect we want to use a different output section name for the first text section. E.g. .init.

And then use .text for the "main" text section.

This would mean that additional declarations of output sections for .text would go into the right place. And this would match the users expectation of what is in .text .

Reply:
Do agree I'll draft up a proposal